### PR TITLE
Updates to fix some bugs in the 'dcos task exec' command.

### PR DESCRIPTION
**Pull request by klueska**

### CODE REVIEW

File: dcos/mesos.py [hunk 1]
```diff
@@ -1315,19 +1314,29 @@ def _input_streamer():
...
-        # exception and break out of this thread.
-        http.post(
-            self.agent_url,
-            data=_initial_input_streamer(),
-            **req_extra_args)
+        # Send an intial "Test" message to ensure that we are able to
+        # establish a connection with the agent. If we aren't we will throw
+        # an exception and break out of this thread. However, in cases where
+        # we receive a 500 response from the agent, we actually want to
+        # continue without throwin an exception. A 500 eror indicates that
```
Comment (tamarrow) [inline dcos/mesos.py]: s/thowin/throwing

Comment (tamarrow) [inline dcos/mesos.py]: s/eror/error

File: dcos/mesos.py [hunk 2]
```diff
@@ -1315,19 +1314,29 @@ def _input_streamer():
...
+        # running. In that case we continue running to allow the output queue
+        # to be flushed.
+        try:
+            http.post(
+                self.agent_url,
+                data=_initial_input_streamer(),
+                **req_extra_args)
+        except DCOSHTTPException as e:
+            if e.response.status_code == 500:
+                container_completed = True
```
Comment (tamarrow) [inline dcos/mesos.py]: this isn't being used.

**Comment by tamarrow** [comment]
@klueska a few nits. looks like you need a rebase

File: dcos/mesos.py [hunk 3]
```diff
@@ -1315,19 +1314,28 @@ def _input_streamer():
...
+        # running. In that case we continue running to allow the output queue
+        # to be flushed.
+        try:
+            http.post(
+                self.agent_url,
+                data=_initial_input_streamer(),
+                **req_extra_args)
+        except DCOSHTTPException as e:
+            if e.response.status_code == 500:
+                pass
```
Comment (tamarrow) [inline dcos/mesos.py]: should we re-raise other exceptions here?

Comment (klueska) [inline dcos/mesos.py]: Yes of course. I must have removed that by accident in my update.

File: dcos/mesos.py [hunk 4]
```diff
@@ -1246,7 +1245,15 @@ def _process_output_stream(self, response):
...
+            # Since upgrading to python 3.5 we now have to read from the
+            # response in chunk sizes of 1 byte in order to avoid blocking
+            # until the connection is closed. Once we upgrade python again, we
+            # should be able to remove this restriction and change it back to
+            # 'chunk_size=None'. See the links below for the bug reports.
+            #
+            # https://github.com/kennethreitz/requests/issues/3421
+            # https://github.com/shazow/urllib3/issues/928
+            for chunk in response.iter_content(chunk_size=1):
                 records = self.decoder.decode(chunk)
```
Comment (tamarrow) [inline dcos/mesos.py]: @klueska I've left a comment on the JIRA about this. Let's try to track it down a little farther before we make this change

File: dcos/mesos.py [hunk 5]
```diff
@@ -1315,19 +1322,28 @@ def _input_streamer():
...
+        # we can't connect to the container because it has already finished
+        # running. In that case we continue running to allow the output queue
+        # to be flushed.
+        try:
+            http.post(
+                self.agent_url,
+                data=_initial_input_streamer(),
+                **req_extra_args)
+        except DCOSHTTPException as e:
+            if not e.response.status_code == 500:
```
Comment (tamarrow) [inline dcos/mesos.py]: 👍

**Pull request merged**